shapiq/games/base.py

```diff
@@ -157,7 +163,106 @@ def is_normalized(self) -> bool:
+            tuple_types = [set(map(type, coal)) for coal in coalitions]
+            if len(tuple_types) > 1:
+                raise TypeError("Elements of tuple must have the same type.")
+            if tuple_types[0] not in [set(), {int}, {str}]:
+                raise TypeError("Tuples must contain either integers or strings.")
+
+            # check that string tuples are only used if player names are provided
+            if self.player_name_lookup is None and tuple_types[0] == {str}:
+                raise TypeError("Player names have to be provided to evaluate string tuples.")
+
+            # convert strings to integers
+            if tuple_types[0] == {str}:
+                coalitions = [
+                    tuple([self.player_name_lookup[name] for name in coal]) for coal in coalitions
+                ]
+
+            # convert list of tuples to one-hot encoding
+            coalitions = transform_coalitions_to_array(coalitions, self.n_players)
+            return coalitions
+        elif isinstance(coalitions, tuple):
+            # Check that tuple has consistent types
+            tuple_types = set(type(player) for player in coalitions)
+            if len(tuple_types) > 1:
+                raise TypeError("Elements of tuple must have the same type.")
+            if tuple_types not in [set(), {int}, {str}]:
+                raise TypeError("Tuple must contain either integers or strings.")
+
+            # check that string tuples are only used if player names are provided
+            if self.player_name_lookup is None and tuple_types == {str}:
+                raise TypeError("Player names have to be provided to evaluate string tuples.")
+
+            # convert strings to integers
+            if tuple_types == {str}:
+                coalitions = tuple([self.player_name_lookup[name] for name in coalitions])
+
+            # convert tuple to one-hot encoding
+            coalitions = transform_coalitions_to_array([coalitions], self.n_players)
+            return coalitions
+        elif isinstance(coalitions, np.ndarray):
+            if len(coalitions) == 0:
+                raise ValueError("The array of coalitions is empty.")
+            if coalitions.ndim == 1:
+                if len(coalitions) < self.n_players or len(coalitions) > self.n_players:
+                    raise ValueError(
+                        "The array of coalitions is not correctly formatted."
+                        f"It should have a length of {self.n_players}"
+                    )
+                coalitions = coalitions.reshape((1, self.n_players))
+            if coalitions.shape[1] != self.n_players:
+                raise TypeError(
+                    f"The number of players in the coalitions ({coalitions.shape[1]}) does not match "
+                    f"the number of players in the game ({self.n_players})."
+                )
+            return coalitions
+        elif isinstance(coalitions, str):
```

Comment:
This logic could also work with an integer (i.e. just the id of a player) dunno if we want this actually. Could be maybe misleading if users mean "get me the first coalition". Maybe this would be better served with a _getitem_ or something.